meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1634 @@
+            roughness_map_node.location = -400, -200
+            roughness_map_node.image = bpy.data.images.load(roughness_map_path)
+            roughness_map_node.image.colorspace_settings.name = 'Non-Color'
+            node_tree.links.new(roughness_map_node.outputs["Color"], principled_node.inputs["Roughness"])
+
+            # Add a texture node for the normal map
+            normal_map_node = nodes.new(type="ShaderNodeTexImage")
+            normal_map_node.location = -800, -600
+            normal_map_node.image = bpy.data.images.load(normal_map_path)
+            normal_map_node.image.colorspace_settings.name = 'Non-Color'
+            noamel_map_adjustment = material.node_tree.nodes.new('ShaderNodeNormalMap')
+            noamel_map_adjustment.location = -400, -600
+            node_tree.links.new(normal_map_node.outputs["Color"], noamel_map_adjustment.inputs["Color"])
+            node_tree.links.new(noamel_map_adjustment.outputs["Normal"], principled_node.inputs["Normal"])
+
+            '''
+            # TODO add AO
+            # Add a texture node for the ambient occlusion map
+            ambient_occlusion_node = nodes.new(type="ShaderNodeTexImage")
+            ambient_occlusion_node.location = -400, 200
+            ambient_occlusion_node.image = bpy.data.images.load(ao_map_path)
+            ambient_occlusion_node.image.colorspace_settings.name = 'Non-Color'
+            node_tree.links.new(ambient_occlusion_node.outputs["Color"], principled_node.inputs["Ambient Occlusion"])
+            #'''
+            
+            '''
+            # TODO add thickness
+            # Add a texture node for the thickness map
+            thickness_map_node = nodes.new(type="ShaderNodeTexImage")
+            thickness_map_node.location = -400, -200
+            thickness_map_node.image = bpy.data.images.load(thickness_map_path)
+            thickness_map_node.image.colorspace_settings.name = 'Non-Color'
+            node_tree.links.new(thickness_map_node.outputs["Color"], principled_node.inputs["Transmission"])
+            #'''
+
+            # Set the subsurface properties
+            principled_node.inputs["Subsurface"].default_value = 0.001
+            principled_node.inputs["Subsurface Color"].default_value = (1, 0, 0, 1)
+
+            # Link the output of the Principled BSDF node to the material output
+            output_node = nodes.new(type="ShaderNodeOutputMaterial")
+            output_node.location = 400, 0
+            node_tree.links.new(principled_node.outputs["BSDF"], output_node.inputs["Surface"])
+
+        else:
+            texture_name = context.window_manager.smpl_tool.texture
+
+            if (len(obj.data.materials) == 0) or (obj.data.materials[0] is None):
+                self.report({'WARNING'}, "Selected mesh has no material: %s" % obj.name)
+                return {'CANCELLED'}
+
+            # Find texture node
+            node_texture = None
```

Comment:
A little advanced, but this can be written:

    node_texture = next(node for node in nodes if node.type == 'TEX_IMAGE', None)